src/refract/dsd/Object.cc

```diff
@@ -58,6 +59,23 @@ Object& Object::operator=(Object rhs)
     return *this;
 }
 
+void Object::push_back(std::unique_ptr<IElement> el)
+{
+    assert(el);
+
+    if (!empty())
+        if (auto mbr = TypeQueryVisitor::as<const MemberElement>(el.get()))
+            if (!mbr->empty() && mbr->get().key())
+                if (auto str = TypeQueryVisitor::as<const StringElement>(mbr->get().key()))
+                    if (!str->empty()) {
+                        auto it = find(str->get().get());
+                        if (it != end())
+                            erase(it);
+                    }
```

Comment:
This cascade is little bit confusing while reading

Reply:
It just checks whether we are adding a MemberElement with StringElement key so we ensure uniqueness - if the key is already encountered, it shall be removed. Specific suggestions how to rewrite this?